src/bp/sdk/botpress.d.ts

```diff
@@ -1360,12 +1363,62 @@ declare module 'botpress/sdk' {
     node: string
   }
 
+  export type FormDataField = undefined | number | boolean | string | FormData[]
+
   export interface FormData {
     id?: string
     contentType?: string
-    [key: string]: undefined | number | boolean | string | FormData[]
+    [key: string]: FormDataField
   }
 
+  interface FormOption {
+    value: string
+    label: string
+    related?: FormField
+  }
+
+  interface FormContextMenu {
+    type: string
+    label: string
+  }
+
+  export interface FormDynamicOptions {
+    endpoint: string // an enpoint to call to get the options
+    path?: string // used with _.get() on the data returned by api to get to the list of items
+    valueField: string // field from DB to map as the value of the options
+    labelField: string // field from DB to map as the label of the options
+  }
+
+  export interface FormAdvancedSetting {
+    key: string
+    label: string
+    type: string
+    moreInfo?: FormMoreInfo
+  }
+
+  export interface FormField {
```

Comment:
yeah it was the only way to access the types from different places, we might need to revisit at some point

Reply:
@EFF Some of those typings are referenced in the SDK (prompts, variables, conditions) so they must be in this file. Otherwise we'd need to import typings from elsewhere and the sdk file wouldn't be standalone anymore. Typing them as any is not an option.